addon/services/items-service.js

```diff
@@ -15,77 +26,101 @@ export default Service.extend(serviceMixin, {
   /**
    * Item Search
    */
-  search (form, portalOpts) {
-    const qs = this.encodeForm(form);
-    const urlPath = `/search?${qs}&f=json`;
-    return this.request(urlPath, null, portalOpts);
+  search (form/* ,portalOpts */) {
+    const session = this.get('session.authMgr');
+    return searchItems({
+      searchForm: form,
+      authentication: session,
+      fetch
+    });
   },
 
   /**
    * Get the item json
    */
-  getById (itemId, portalOpts) {
-    const qs = this.encodeForm(this.get('defaultParams'));
-    const urlPath = `/content/items/${itemId}?${qs}`;
-    return this.request(urlPath, null, portalOpts);
+  getById (itemId/* ,portalOpts */) {
+    const session = this.get('session.authMgr');
+    return getItem(itemId, { authentication: session, fetch });
   },
 
   /**
    * Get the `/data` as json. If nothing is returned by AGO
    * and empty object (`{}`) will be returned by this call
    */
-  getDataById (itemId, portalOpts) {
-    const qs = this.encodeForm(this.get('defaultParams'));
-    const urlPath = `/content/items/${itemId}/data?${qs}`;
-    return this.request(urlPath, null, portalOpts);
+  getDataById (itemId/* ,portalOpts */) {
+    const session = this.get('session.authMgr');
+    return getItemData(itemId, { authentication: session, fetch });
   },
 
   /**
    * Update an existing item
    * will update the `/data` if the `.text` value is present
    */
-  update (item, portalOpts) {
-    const urlPath = `/content/users/${item.owner}/items/${item.id}/update?f=json`;
-    return this._post(urlPath, item, portalOpts);
+  update (item/* ,portalOpts */) {
+    const session = this.get('session.authMgr');
+    return updateItem({
+      item,
+      owner: item.owner,
+      authentication: session,
+      fetch
+    });
   },
 
   /**
-   * Create a new item in a particular folder
+   * Create a new item in a particulnpm run lint:jsar folder
```

Comment:
looks like your keyboard focus was in the wrong window when you typed `npm run lint:js` ;)